source/common/http/header_map_impl.h

```diff
@@ -52,6 +52,9 @@ public:
  */
 class HeaderMapImpl : NonCopyable {
 public:
+  HeaderMapImpl(const uint32_t max_headers_kb = UINT32_MAX,
```

Comment:
I'm not sure if the default values here are needed or not. If they cause too much of a code-churn at the moment then they should be kept, otherwise, it will be better to remove them.

Reply:
Yeah, without the default, it will cause too much code-churn in the test code.  It's also important for the StaticEmptyHeaders used in local reply as here:  https://github.com/envoyproxy/envoy/blob/edebad172551bb5029a5027ae5fb927510ca2696/source/common/http/header_map_impl.h#L638

Reply:
is `UINT32_MAX` a good default though? I am wondering can we use `DEFAULT_MAX_REQUEST_HEADERS_KB ` and `DEFAULT_MAX_HEADERS_COUNT` that are in envoy/http/codec.h?

Reply:
UINT32_MAX seems ok. It means there is no limit, which preserves the behavior.

Reply:
ok, if `no limit` is the behavior we want to preserve, we can probably also use absl::optional for those limits.

Either works for me, no strong preference.

Reply:
Yeah, no limit is the behavior.  Let me stay with UNINT32_MAX to keep it simple.

Reply:
> ok, if `no limit` (i.e, optional) is the behavior we want to preserve, we can probably also use absl::optional for those limits.
> 

Using optional makes sense from a maintainability perspective. However, it has some memory overhead, so if there's a valid default value, then it should be used.

Reply:
Thanks for the comments. I think default to UNINT32_MAX  implies no limit.